src/bots/court.js

```diff
@@ -1,338 +1,301 @@
-    "EXECUTION": 4
-}
-
-
-module.exports = async (web3, mongoClient, courtAddress, archon) => {
-  // Instantiate the contracts.
-  const courtInstance = new web3.eth.Contract(
-    _court.abi,
-    courtAddress
-  )
-
-  const policyRegistryInstance = new web3.eth.Contract(
-    _policyRegistry.abi,
-    '0xCf1f07713d5193FaE5c1653C9f61953D048BECe4'
-  )
-
-  // connect to the right collection
-  await mongoClient.createCollection(COURT_MONGO_COLLECTION)
-  const db = mongoClient.collection(COURT_MONGO_COLLECTION)
+module.exports = async ({ web3, court, policyRegistry, mongoCollection, archon, webhookUrl }) => {
+  const courtAddress = court.options.address;
 
   // get our starting point
-  let lastBlock = process.env.START_BLOCK
-  let currentBlock = process.env.START_BLOCK
-  let votingDisputes = []
-  let appState = await db.findOne({'courtAddress': courtAddress})
+  let lastBlock = process.env.START_BLOCK;
+  let currentBlock = process.env.START_BLOCK;
+  let votingDisputes = [];
+  const appState = await mongoCollection.findOne({ courtAddress: courtAddress });
   if (appState) {
-    lastBlock = appState.lastBlock
-    votingDisputes = appState.votingDisputes || []
-  }
-  else {
+    lastBlock = appState.lastBlock;
+    votingDisputes = appState.votingDisputes || [];
+  } else {
     // if starting from scratch we can go from the current block
-    await db.insertOne({'courtAddress': courtAddress, 'lastBlock': currentBlock})
+    await mongoCollection.insertOne({ courtAddress: courtAddress, lastBlock: currentBlock });
   }
-  votingDisputes = [ ...new Set(votingDisputes) ]
+  votingDisputes = [...new Set(votingDisputes)];
 
   while (true) {
-    await delay(process.env.DELAY_AMOUNT)
-    currentBlock = await web3.eth.getBlockNumber()
-    const drawEvents = await courtInstance.getPastEvents('Draw', {
+    await delay(process.env.DELAY_AMOUNT);
+    currentBlock = await web3.eth.getBlockNumber();
+    const drawEvents = await court.getPastEvents("Draw", {
       fromBlock: lastBlock,
-      toBlock: 'latest'
-    })
+      toBlock: "latest",
+    });
 
     if (drawEvents.length) {
-      const jurorsForDisputes = await getDrawnJurorsByDispute(drawEvents || [])
-      for (let disputeID of Object.keys(jurorsForDisputes)) {
-        for (let juror of jurorsForDisputes[disputeID]) {
-          const address = juror.address
-          console.log("SENDING DRAW EMAIL TO " + juror.address + " IN CASE " + disputeID)
-          await axios.post('https://iu6s7cave4.execute-api.us-east-2.amazonaws.com/production/event-handler-court-emails',
-            {
-                "event": "Draw",
-                "_disputeID": disputeID,
-                "_appeal": juror.appeal,
-                "_address": juror.address
-            }
-          )
+      const jurorsForDisputes = await getDrawnJurorsByDispute(drawEvents || []);
+      for (const disputeID of Object.keys(jurorsForDisputes)) {
+        for (const juror of jurorsForDisputes[disputeID]) {
```

Comment:
~This is not a problem introduced inside this pull request, but while we are on it:~

~This loop sends an email for each draw, however, draws represent votes, not jurors, so if a juror is drawn 5 times, this will send 5 emails. We can fix it by processing draws to build a temporary set of drawn jurors, then send emails to this set instead.~

All good, `getJurorsInCurrentRound()` handles it already, I didn't notice.